lib/IntervalJitter.js

```diff
@@ -0,0 +1,99 @@
+const EventEmitter = require('events');
+/**
+ * An interval timer with jitter
+ * @example
+ * const j1 = new IntervalJitter()
+ * j1.start({interval: 1000, jitter: 100}, (t) => { console.log(`hello j1, this was delayed ${t}ms`) })
+ * setTimeout(() => { j1.stop() }, 5000)
+ * @example
+ * const j2 = new IntervalJitter()
+ * j2.on('interval', (t) => { console.log(`hello j2, this was delayed ${t}ms`) })
+ * j2.on('started', () => { console.log('j2 started') })
+ * j2.on('stopped', () => { console.log('j2 stopped') })
+ * // Start with initial "interval" firing at somewhere between 0 ~ 500ms
+ * // After that, fire somewhere between 900 ~ 1100ms
+ * j2.start({interval: 1000, jitter: 200, firstInterval: 0, firstJitter: 500})
+ * setTimeout(() => { j2.stop() }, 5000)
+ */
+class IntervalJitter extends EventEmitter {
+    constructor () {
+        super()
+        this.interval = 1000
+        this.jitter = 1000
+        this.firstInterval = null
+        this.firstJitter = null
+        this.callback = null
+        this.#init()
+    }
+    #stopped = true
```

Comment:
If you are going to use private class members you'll need to update `ecmaVersion` to `latest` in `.eslintrc`